### DISCUSSION TITLE
MMO Hackathon

### DISCUSSION THREAD

**Discussion by adamhannigan** [sunflower-land/sunflower-land#2575] [Ideas]
Welcome Bumpkins, to the first ever team Hackathon.

On the 23rd of May, our trusted engineers will embark on a 3 day hackathon in attempts to build and launch a fully-fledged MMO that lets players explore, connect, quest and talk with other players.

<img width="1416" alt="Screen Shot 2023-05-18 at 7 42 38 am" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/34cd2b16-5c24-4fb7-9584-185c8b0d6079">

## Sunflower Isles

The MMO will take place on one of the larger islands in the Sunflower Isles. Here players will be able to venture into different buildings, areas and unlock the secrets of Sunflower Land's history.

## What is the goal of an MMO style game?

It is no secret than many "Metaverse" game's turn into barren wastelands, so why are we building this?

We view MMOs as a social layer that should only be added once the game is a fun and engaging standalone experience. We are now at that point. 

In particular, we are adding MMO capabilities to:

- Amplify the social experience 
- Show your rare items and wearables
- Share experiences and help newcomers (in-game)
- Strengthen the community

## Features

Although we are a small team (4 engineers), we are going full guns blazing over this 3 days to build out this feature.

For the MVP, we are looking to launch:

- Walk, talk and interact with other Bumpkins
- NPC interactions
- Lore, secrets and background history
- Multiple servers
- Friend Lists
- Auction House
- Leaderboard
- Museum (Fan Art)

Is there something you would like to see in this feature? Please comment any suggestions you have ❤️ 

## Where to watch us?

We will be doing live streams on our [Twitch](https://www.twitch.tv/0xsunflowerstudios) & giving live updates on our [TikTok](https://www.tiktok.com/@sunflower_land.studios)

These streams will be interactive and your feedback will help shape the success of this feature. During the streams, we will give viewers special early access to explore land as it is being built.

**Comment by digital-farms** [comment]
**"LOVE SFL" statues:**
![LOVE_SFL_NFT](https://user-images.githubusercontent.com/105886943/210788552-3f9185dc-730e-4d72-a28d-3c217511889e.gif) - GIF
![LOVE_SFL_NFT_static](https://user-images.githubusercontent.com/105886943/210788558-ddd4d231-86db-43bd-9512-95d502ac072c.png) - Static

![LOVE-SFL-STATUE-ON-HELIOS](https://user-images.githubusercontent.com/105886943/210786457-c74883e1-67ea-443b-85e4-19480c774a25.png)

`land id: 21637`
`discord: KOTOB#4391`

**Reply by sabryder** [reply]
i would love this to be add into them game for craft. amazing art. the best one ive seen for decoration😍😍😍❤❤

**Comment by Chucrutty** [comment]
wish i could hit something and advance like dungeons.... like lvl = power  as diablo or some thing like that .. maybe use skelots for this, some drop chance of rare itens

**Reply by paduv** [reply]
Yeah it would be nice, but i doubt 3 days will be enough time to make it right :)

Also the animation involved in Diablo-like stuff might be too demanding for a side project for a small team.

But maybe turn based combats would be more doable (ala Final Fantasy 1-9 etc) for future development

**Reply by Alaindrea** [reply]
I want a whack-a-skeleton game! like whack-a-mole but no moles will be harmed

**Reply by adamhannigan** [reply]
Nice ideas!

It would be nice to make it farming themed somehow. Or perhaps moonseekers ;)

**Comment by AxieFan1** [comment]
Some suggestions:

1) In near future, include real life stores, that players can order from, if that store has a presence locally... like pizza, fashion etc.
2) Organize one-off events like stand-up shows, gigs in association with partner performers.
3) Involve socially-oriented NGOs, if they would be interested in having a stand there.
4) A barter house, where players can exchange stuff upon successful negotiations, securely and promptly.
5) Ability of non-SFL players to visit and experience the ambience, without the option to chat or do any unacceptable acts.
6) Crypto news room, with latest relevant news flows.
7) Cross-game resources usage, that allows more exposure and switch over from new players.

land id: 6530
discord: AxieFan1#5709

**Reply by adamhannigan** [reply]
Awesome ideas @AxieFan1  🏆

**Comment by Alaindrea** [comment]
Create the ability for Bumpkins to set up a farm stand in that area where they can sell resources or trade with other bumpkins. The bumpkin has to stay with their farm stand, so there is a limit to how long each stand is open. Maybe they pay a small fee to set up their stand?

**Reply by adamhannigan** [reply]
This is really cool - we may need a separate area for this - The "Farmer's Market".

Perhaps players could pay for a premium location on the map

**Comment by vitt0** [comment]
It would be good to bring back the Town Hall and also the NPC that use to say the SFL current supply, as funny as it sounds I see guys asking "when the next halvening?" pretty often 😄 

![image](https://github.com/sunflower-land/sunflower-land/assets/11968934/ae73bfbf-d62a-4db5-b74f-6615ff904743)

**Reply by adamhannigan** [reply]
Hahah, love it 👌

**Comment by rodolfoddias** [comment]
Please, bring the #2203 proposal to life! Would love to have some interactions with some NPCs and develop relationships with them just like in Harvest Moon and Stardew Valley!

**Reply by adamhannigan** [reply]
Definitely! This is the start of that. 

We will have more interactive dialogue - relationship building will take a little longer to build however

**Comment by NilzGone** [comment]
Maybe too much for 3 days but I'd love you forever if there were some combat that could be built upon, even if it's just a rat that takes a random number of clicks to kill and each click does a retaliation dmg until it dies, the bumpkin lost health we'd have to feed it to heal it, they can never actually die just go down to 1hp and then can't kill anymore, and is a separate thing from xp, eg you wouldn't heal and get xp at the same time. loot from the rat would be a few seeds that he's stolen from, maybe a small chance for a fruit or maybe a TI drop that a rat could steal, maybe the sea cucumber?

Also maybe the ability to use an item on someone, the main reason for this is I want everyone to get a Christmas cracker for Christmas and then we can pull it with another person (BLUE PARTY HAT PLS ;)), clearly stolen from RS but in wow as well there are some items you can use on another player like snowballs etc, maybe it could be fun and fit, I'm not too sure

**Reply by adamhannigan** [reply]
Awesome ideas 👌 

In terms of combat we would love to explore these ideas - In an MMO arena this is quite technically challenge though. Hopefully we will see this soon.

The idea of using items on other people is awesome. We want the space to be interactive, so players should be able to influence each other.

**Comment by edgardopouch** [comment]
I would like to see more buildings in sunflowerland.
in which the resources obtained this time are spent.
A castle 🏰 to burn stone. Where npc with special functions live.
The formation of clans could also be enabled. and monument assembly competitions.
after the clans that the villages can be united and be able to form kingdoms and later war between clans.
coming soon sunflower kingdoms?

**Reply by adamhannigan** [reply]
Lots more buildings coming. Sunflorian Kingdom is planned for the map ;) 

We already have some awesome art ready! @edgardopouch

**Comment by gven09** [comment]
Hello,
I would like to suggest a ship for each land which crafted with iron wood ect. Also you can travel other islands or a unknown discovery.

**Comment by edmilsonventura13** [comment]
hello, I would like to suggest the interaction of farmers with some games:

-npc/dice games (bet on SLF to guess the dice number). gains in SFL or resources.
-npc/charade games with several farmers trying to guess (suggestion: charades involving animals would be a good one because we all have the references, because charades involving subjects in general will not all have the reference since Sunflower Land has players from several places in the world) .
-npc/roulette game involving betting.
-Auction of mystery boxes (may contain random high value or low value items).
-for the winners some medal or trophy exclusive to the event (decoration).

it would also be good to be able to forge some exclusive item of the event (decorative) items with low value and high value for all types of players
-exclusive items for humans, goblins and pirates. (in my opinion there is a lack of a race mechanic)

raffle of some ticks (down breaker ticket)

I could suggest more suggestions but I believe that for a three-day event it would be too much, as implementing an MMO takes work. I'm a huge MMORPG fan if Sunflower Land ever goes down this path it would be great as it has great potential.

thanks

edymonth#3020

**Reply by rodolfoddias** [reply]
Sunflower Land Casino!

**Reply by edmilsonventura13** [reply]
algo comum em eventos. :D

**Comment by VZozul** [comment]
1. Lucky tree
2. Motivation and leaderboard
3. Quiz time

I say right away that these ideas were taken from testing another project, it can be safely implemented, they refused to implement it. Yes, and you can do a little in your own way, just take the idea as a basis.

1. Lucky tree.
It can make people get together at the same time and have fun. And it's really fun. The idea is very simple, trees are reset 2 times a day. Here you can adjust the number of times depending on the number of players who will participate. You can do it once a day, and 10 times a day. Given the time difference for everyone, you need to do 2-3 times a day. Well, or if once a day, then choose a time that is convenient for everyone. For example, I'm from Ukraine, and the reset time in the game is 3 o'clock in the morning. Getting up so early every day is also not very good. Therefore, everyone needs something in between or 2 times a day. The bottom line is that you as developers will not lose money on the lottery, you can choose to stay at 0, or earn not much. If I were you, I would stay at 0, but due to the fact that people will enter the game, from here you yourself understand what follows. You will calculate the tokenomics of the rewards yourself. I understand that it will be quite difficult to implement this, but it's worth it! The way it was implemented there, you enter the building, outside the building and inside there is a board with awards, it is very important to make sure that you have to go there! If you constantly see what rewards have already been taken, then you can stop in time. And when you have to run around to see the awards, then during this time someone can pick up the awards, and when you collect it, you don’t know or they were taken away, it's much more interesting. Further into this building there are a lot of trees, you go into the tree and choose a toy and pay to open it, and immediately then it shows you what kind of reward is there, or there is no reward) It is clear that when all the normal rewards are taken away, no one will collect further, this too you need to calculate so that you do not lose. I'll post screenshots below. Of course, you can do it without a building, just make a passage to the territory with Lucky trees, put a Board with awards at the beginning of the passage, make trees different, toys different, but just cutting down a tree will not work, too fast. You need to go into the tree, select something there and click. And believe me, it's really fun when hundreds of people run from the trees to the Board with awards back and forth. Someone stands near the board of rewards and waits for the best moment, when there are more chances. The screenshots also have a table with the awards that were there. As we can see, there are few big awards, and many small ones without awards.

2. Motivation, here is a separate story that needs to be paid attention to. The motivation system should be a leaderboard table. With the beginning of the season, a leaderboard opens, by the end of the season, the first 10 places get something. What do we get points for ... There are many options, every day opening a daily chest - 1 point, for every 5 day strike - 2 points. Quiz time - I'll describe it below. For each correct answer you get points, with the reward system that will be there, I will also describe below. For each completed seasonal quests - 1 point, motivates even after you have collected Season tickets for the SFT that you have chosen, you will continue to do quests to collect points! It will also be possible to introduce seasonal achievements, as is, but broken down into smaller ones. And you can still get points for a lot, here you need to think, but the main thing to which we are moving is to the Lucky Tree. This is the main RNG that adds excitement! Since in the Lucky tree, for each reward taken, we get Points. For example, for each discovery of a toy, we pay 1 SFL and 1 point is removed from us, if we win 10 SFL, we also get 10 points. 3 SFL - 3 points and so on. That is, roughly speaking, if we open 10 times and get nothing, then we will have minus 10 points, in the minimum reward where we get 1 SFL and 1 point, we essentially lose nothing, but if we knocked out 100 SFL and got 100 points, there will be a lot of happiness)  And at the expense of those who open and do not receive anything, Lucky Tree tokenomics is built without loss for developers.

3. Quiz time! A certain number of times a day, at a random time, an announcement appears that it's Quiz time! We can see it wherever we are. Then we sail to a new island, there we run to the Quiz Time Board, or to the character an analogue of how we take quests, quickly press and answer the question. The first one to answer correctly gets 10 points, the second 9 and so on, the 10th gets 1 point and everyone else can answer the question at any time until the next quiz, but only get 1 point. Whoever is faster will get more.
Questions in the Quiz may be different, where we choose the answer from the listed. And this opens a lot of doors for sponsors. Every sponsor wants to be interested in their project, then it is in their interest to have their questions about their project appear in Quiz Time. Here you will have something to think about cooperation. In addition to points, you can receive awards in the SFL, here you can take money from sponsors. The first one to answer correctly will receive 10 points and 1 SFL, the second 9 points and 0.9 SFL, and so on, the rest will receive 1 point and 0.1 SFL. This is for example.  

Below are screenshots to roughly see how it was there.

![111](https://github.com/sunflower-land/sunflower-land/assets/103659539/26d2742b-830c-49ce-9bd1-4ebb16fa2c96)
![RELOAD](https://github.com/sunflower-land/sunflower-land/assets/103659539/abd8042b-c1b1-4576-896e-ca82e2f30c0d)
![LUCKY](https://github.com/sunflower-land/sunflower-land/assets/103659539/3666413d-186a-4c34-96d1-492b33f95e6f)
![5](https://github.com/sunflower-land/sunflower-land/assets/103659539/61991141-674e-4b00-ad04-12ecf329eda4)
![RIGHT](https://github.com/sunflower-land/sunflower-land/assets/103659539/e420db26-aafc-451f-8a4a-9f8a4ed8759f)
![6](https://github.com/sunflower-land/sunflower-land/assets/103659539/50e6f93f-9979-47a5-8449-f446321a12ab)
![4](https://github.com/sunflower-land/sunflower-land/assets/103659539/88731419-f56b-4d7e-a9cd-8699e8f8f2b5)
![7](https://github.com/sunflower-land/sunflower-land/assets/103659539/61b96a23-1517-40b3-8fc5-a522c8905ccf)

**Reply by adamhannigan** [reply]
Great ideas!

These would really encourage players to get together. The Lucky Tree sounds very exciting for me - I'd love to have some sort of daily drops where players get excited and gather around to get some items

**Reply by VZozul** [reply]
Give me a couple of days, I'll think about it, there are a couple of new ideas on how to make the Lucky Tree better. I'll write it all down and show for example how it will look.
Regarding the LeaderBoard system. In my opinion this is worth paying attention to. It's just that there will be many different activities in the "world" location, and there are 2 options here, you can separate these activities, or make a common Leaderboard with points, and receive rewards at the end of the season. The only plus in this is to make motivation for activity in everything, I described this above. But given that there will be an auction, the motivation to complete quests until the last day is if the auction is held at the end of the season. Then maybe there is no point in combining all the activities into one leaderboard with a point system. There is another option to make an auction not for seasonal tickets, but for points that will be earned in all activities. Then there will be motivation to participate in all activities. But anyway, I have a great idea how to make Lucky Tree motivation! A couple of days and I will write in detail.

**Comment by dinryl84** [comment]
It woud be great if you can make another isle where battles between human and goblin takes place. A strategy game like game of the generals. Wherein your the objective of the game is to capture enemy's flag but in this case sunflower.. pumpkin upto beetroot are private soldier. Parsnip upto kale are your officers, wherein officers can easily defeat the private. and officers can be taken down by spy and spy can only be taken down by private... i think you can easily program this game in 3 days span. You can tweak this game by adding timer. If the flag was not captured during the alloted time the winner will be determine base on the rank he captured. If your intetested just google the game of generals it is a board game. This shoud be p2p game and sfl or crops should be at stake...

Here is the rule of the game for your reference.
https://www.geekyhobbies.com/game-of-the-generals-aka-salpakan-review-and-rules/

 I think the farmer would love it after all SFL game involves strategy even if we just planting...

Land no. 68138

I really hope you would consider this...

**Reply by adamhannigan** [reply]
Great idea!

We are trying to give enough flexibility for our open source developers to build these functions into the game 😄 

This would be really cool

**Comment by future2back** [comment]
Please consider the option to moving using ASDW keys.

Thanks

**Reply by TheDcol** [reply]
I agree, more pc friendly :D

**Reply by future2back** [reply]
You might also consider an on-screen joystick for mobile players.

![exampleJoystick](https://github.com/sunflower-land/sunflower-land/assets/74167741/43f1e9af-c653-4652-ae31-de37e1b56bd2)


Also using 'ENTER' key to shortcut chat(would be nice adding the possibility to whisper a specific player and filtering chat) 
Eg:
{command} {farm} {message}

In case of a specific player (Orange)
`/whisper #1 Hello Adam!` 
`/whisper #2 Hello Romy!`

In case of a Guild (Purple)
`/guild Hello Clan!`

Case of co-op (Blue)
`/party Hello Team!`

Case of world (Gray)
`/world Hello World!`

Case of local (White)
`/local Hello Friend!`


## Admin cases

Case of welcome (Yellow)
`/welcome  You ll always see this message in center of display for a few seconds when joining Sunflower Land server, welcome and enjoy the adventure!`

![screenLand](https://github.com/sunflower-land/sunflower-land/assets/74167741/df64270e-d81f-414e-bf9d-96425c130fe0)


Case of global (Red)
`/global Hello Sunflower land players, The devs are on the city right now, join us! Run this message will disappear in a few seconds.`

![screenLand](https://github.com/sunflower-land/sunflower-land/assets/74167741/3fae0a1b-6edd-44e2-8df0-b650864aebaf)




Then msg terminal close after sending with 'ENTER' key.

Commands can be in an autocomplete drop-down list and save the most recent information.
Chat prompt could be separated into 3 blocks:
`|Command|ID|Message|`

User could set it manually or using commands.
command with no message like: 
`/local`
can be used to save a state without sending a message.

Thank you.
